eole/utils/misc.py

```diff
@@ -74,6 +74,51 @@ def use_gpu(config):
     )
 
 
+def clear_gpu_cache():
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
+    if torch.backends.mps.is_available():
+        torch.mps.empty_cache()
+
+
+def get_device_type():
+    if torch.cuda.is_available():
+        return "cuda"
+    elif torch.backends.mps.is_available():
+        return "mps"
+    else:
+        return "cpu"
+
+
+def get_device(device_id=None):
+    if torch.cuda.is_available():
+        if device_id is not None:
+            return torch.device(f"cuda:{device_id}")
+        else:
+            return torch.device("cuda")
+    elif torch.backends.mps.is_available():
+        return torch.device("mps")
+    else:
+        return torch.device("cpu")
+
+
+def get_autocast(enabled=True, device_type="auto"):
+    if not enabled:
+        return nullcontext()
+
+    if device_type == "auto":
+        device_type = get_device_type()
+
+    if device_type == "cuda":
+        return torch.cuda.amp.autocast()
+    elif device_type == "mps":
+        return torch.amp.autocast(device_type="mps")
+    elif device_type == "cpu":
+        return torch.cpu.amp.autocast()
+    else:
```

Comment:
is this possible?

Reply:
No, on reflection, shouldn't be. Will always have CPU.